### ISSUE TITLE
When I chose @pattern the selected information must be filled, in fact it is not

### ISSUE THREAD

**Issue by TLLABRES** [voyages-sncf-technologies/hesperides-gui#95]
I've put a field in an exiting module with the @pattern, I have saved my platform with an empty field. In fact i expect to fill it with a mathing text according the chosen pattern

**Comment by stidiovip** [comment]
This can be fixed in two ways : 
1. Customize the ng-pattern directive for checking empty pattern value ( not really efficient for the API)
2. Make hesperides backend check pattern on data save. ( this could be better solution )

**Comment by TLLABRES** [comment]
I agree with de 2nd proposed soluce, it should be nice to process with a message which explain the expected value fitting the pattern

**Comment by JordanKergoat** [comment]
In fact, the best solution would be to combine both. 
- Adding an ng-pattern will prevent from sending data that are known to be false.
- The backend check is never an option, we do not want corrupted data, in my opinion, this part should be done first.

**Comment by thomaslhostis** [comment]
@JordanKergoat Si la demande est de vérifier qu'une propriété contenant l'annotation @pattern n'est pas vide, il existe l'annotation @required pour ça. Est-ce bien cela ? As-tu un exemple ?

**Comment by Lucas-C** [comment]
Vu avec Jordan, j'ai reformulé le nom de l'_issue_ pour être plus clair
@thomaslhostis : côté backend, est-ce vérifié désormais ?
Ou bien est-ce que des valeurs qui ne "matchent pas" le `@pattern` sont acceptées par souci de retrocompatibilité ?

**Comment by Lucas-C** [comment]
En lien avec:
- https://github.com/voyages-sncf-technologies/hesperides/issues/513 : (backend)  Valider les annotations @required et @pattern lors de la valorisation en tenant compte des références de propriétés
- https://github.com/voyages-sncf-technologies/hesperides-gui/issues/175  Les valorisations de propriétés ne respectant pas le pattern ne sont pas sauvegardé dans le ngModel AngularJS, sans qu'aucun message n'en informe l'utilisateur #175

**Comment by stidiovip** [comment]
Okay. Fine !
